src/Symfony/Component/Messenger/Middleware/SendMessageMiddleware.php

```diff
@@ -36,7 +37,9 @@ class SendMessageMiddleware implements MiddlewareInterface
     public function __construct(SendersLocatorInterface $sendersLocator, EventDispatcherInterface $eventDispatcher = null)
     {
         $this->sendersLocator = $sendersLocator;
-        $this->eventDispatcher = class_exists(Event::class) ? LegacyEventDispatcherProxy::decorate($eventDispatcher) : $eventDispatcher;
+        $this->eventDispatcher = class_exists(Event::class) && $eventDispatcher instanceof SymfonyEventDispatcherInterface
```

Comment:
Is the `class_exists()` check necessary?  If SymfonyEventDispatcherInterface is defined, then the package is present.  If the package is present, the Event class necessarily will exist, no?  The two checks are effectively redundant.

Reply:
> If the package is present, the Event class necessarily will exist, no?

No, that class has been removed in 5.0. We're checking if we have an EventDispatcher 4.4 here. This hot mess is part of Symfony's migration to PSR-14. On the 6.0 branch, we were finally able to remove it.

Reply:
Ahhh...  Got it, OK.